access/types/stdsync.py

```diff
@@ -143,18 +143,18 @@ def createForm(request, course, exercise, post_url):
     if form.is_valid():
         (points, error_groups, error_fields) = form.grade()
         points = pointsInRange(points, exercise["max_points"])
+        if not 'rejected' in error_groups:
```

Comment:
At first sight, it looks suspicious that `form.is_valid()` would pass for a rejected form. We need to compare this to how checks for required fields are handled, since those can similarly lead to rejected forms.

Reply:
So, we suspect that the required fields are handled by the validation provided by the superclass form (Django forms.Form). You should try to implement custom validation methods on the GradedForm class. They would be called by the `form.is_valid()` method (implemented in the Django superclass). The custom validation method would reject the form if the input can not be parsed as an integer in a text field with the int type/comparator.

https://docs.djangoproject.com/en/2.2/ref/forms/validation/

Reply:
Since GradedForm creates the form fields dynamically based on the exercise configuration YAML, the `form.clean` method might be a suitable place for looping over the fields and checking that integer text fields contain valid integers.  Another alternative is to add validators to the text fields (default field validation runs the validators). The field constructor has a validators parameter.

Reply:
It could be even better to use more descriptive form fields such as IntegerField instead of TextField when we expect integer input. IntegerField also validates the input by default, so we would not need custom validation methods for that.